components/Buttons/src/MDCFlatButton.m

```diff
@@ -16,12 +16,23 @@
 
 #import "MDCFlatButton.h"
 
+#import "MaterialShadowElevations.h"
 #import "private/MDCButton+Subclassing.h"
 
 static NSString *const MDCFlatButtonHasOpaqueBackground = @"MDCFlatButtonHasOpaqueBackground";
 
 @implementation MDCFlatButton
 
++ (void)initialize {
```

Comment:
lets move the logic into the common init.

If we need to override the setters in the subclass as you showed me in your experiment lets do that too.

we should also confirm the default values with a unit test.